src/proxy-ui-api/frontend/tests/e2e/specs/ss-backupandrestore.js

```diff
@@ -0,0 +1,270 @@
+      .elements('//table[contains(@class, "xrd-table")]/tr')
+      .count.to.equal(1);
+    backupAndRestoreTab.clearFilterInput();
+
+    // Delete created backup (click cancel first time)
+    browser.waitForElementVisible(
+      `//table[contains(@class, "xrd-table")]//td[text() = "${createdBackupFileName}"]/following-sibling::td//button[contains(@data-test, "backup-delete")]`,
+    );
+    backupAndRestoreTab.clickDeleteForBackup(createdBackupFileName);
+    browser.waitForElementVisible(deleteBackupConfirmationDialog);
+    deleteBackupConfirmationDialog.cancel();
+    browser.expect.element(deleteBackupConfirmationDialog).to.not.be.visible;
+
+    await deleteBackup(browser, createdBackupFileName);
+    browser.end();
+  },
+
+  'Download and import backup': async (browser) => {
+    const mainPage = browser.page.ssMainPage();
+    const settingsTab = mainPage.section.settingsTab;
+    const backupAndRestoreTab = settingsTab.section.backupAndRestoreTab;
+    const backupFileAlreadyExistsDialog =
+      backupAndRestoreTab.section.backupFileAlreadyExistsDialog;
+    const deleteBackupConfirmationDialog =
+      backupAndRestoreTab.section.deleteBackupConfirmationDialog;
+
+    // delete existing backups from test dir
+    const testDataDir = __dirname + browser.globals.e2etest_testdata + '/';
+    const regex = /^conf_backup/;
+
+    fs.readdirSync(testDataDir)
+      .filter((file) => regex.test(file))
+      .map((file) => fs.unlinkSync(testDataDir + file));
+
+    // Navigate to backup and restore page
+    navigateToBackupAndRestorePage(browser);
+
+    // Create backup
+    backupAndRestoreTab.clickCreateBackup();
+
+    // Make sure backup was successfully created
+    browser.assert.containsText(
+      mainPage.elements.snackBarMessage,
+      'successfully created',
+    );
+
+    // Get the backend-generated name of the backup and close snackbar
+    const createdBackupFileNameTextObject = await browser.getText(
+      'xpath',
+      mainPage.elements.snackBarMessage,
+    );
+    const createdBackupFileName = createdBackupFileNameTextObject.value.split(
+      ' ',
+    )[1];
+    console.log('Created backup: ', createdBackupFileName);
+    mainPage.closeSnackbar();
+
+    // Download backupfile and make sure it's in the filesystem
+    backupAndRestoreTab.clickDownloadForBackup(createdBackupFileName);
+    browser.pause(5000);
+
+    browser.perform(() =>
+      browser.assert.equal(
+        fs.existsSync(`${testDataDir}${createdBackupFileName}`),
+        true,
+        `Backup file should exist in: ${testDataDir}${createdBackupFileName}`,
+      ),
+    );
+
+    // Import the created backup from local filesystem (first cancel the operation)
+    backupAndRestoreTab.addBackupToInput(testDataDir + createdBackupFileName);
+    browser.waitForElementVisible(backupFileAlreadyExistsDialog);
+    backupFileAlreadyExistsDialog.cancel();
+    browser.expect.element(backupFileAlreadyExistsDialog).to.not.be.visible;
```

Comment:
Not sure but wouldn't `waitForElementNotVisible(...)` actually wait (10000ms) for the element to not be visible where as `.to.not.be.visible` expects the element to not be visible immediately? Should `waitForElementNotVisible` be preferred in these cases?

Reply:
Sounds reasonable. Changed.